### DISCUSSION TITLE
What is _next/static/chunks/framework.*******.js?

### DISCUSSION THREAD

**Discussion by deependras1110** [vercel/next.js#38462] [Help]
Is this some static file?

**Comment by icyJoseph** [comment]
That's the react core and the react-dom library packaged to be sent to the client, both minified, uglyfied and optimised for production.

**Reply by deependras1110** [reply]
So one of the errors I get after deploying the next.js app on production is this framework.js missing along with webpack.js
So is it the issue with the dependencies, version of react?

**Reply by icyJoseph** [reply]
With this much information the issue could be anything really. 

When next.js does the build it compiles this files iirc. I'm not sure what the actual pipeline looks like, but these end up in the `.next` folder.

I really don't think this has anything to do with next.js or react, rather this ought to be a pipeline error. The question we must answer is, why are these missing? 

If you build locally, can you see them? What does the production deploy do? Are you using a custom webpack config?

**Reply by deependras1110** [reply]
I inspected the application locally, and I can see the webpack.js but not framework.js under Network. However, I am not sure if the application needs to fetch framework.js in local since it's for packaged content into production. 
As far as the custom webpack, yes have custome webpack config

Also, the project with next.js v9 is working fine when in prod, but upgrading to next.js v12, the prod is breaking.

**Reply by icyJoseph** [reply]
> I inspected the application locally, and I can see the webpack.js but not framework.js under Network. 

You should see the structure of `.next` and see if it's there, at the `.next/_next/static/chunks/framework.[hash].js` location.

> However, I am not sure if the application needs to fetch framework.js in local since it's for packaged content into production.

What do you mean here? If you build the application and run it, locally, in a server, the client will still request the framework because the script is part of the initial HTML

<img width="579" alt="Screenshot 2022-07-11 at 23 26 19" src="https://user-images.githubusercontent.com/21013447/178361351-b651958f-7dbd-4cf4-9f02-cc17f74f0f45.png">

> As far as the custom webpack, yes have custome webpack config
> Also, the project with next.js v9 is working fine when in prod, but upgrading to next.js v12, the prod is breaking.

I see, this was good to know. Could it be that there's some issue with the file hashing? What plugins or what does the custom webpack config do? From Next.js v9 to v12, WebPack 5 became enforced, so there might definitely be some breaking changes there, brought by webpack v5.

**Comment by isayanotud** [comment]
/_next/static/chunks/16fce0~0kfr7_.js
